chore: fail tests if the test sarif reports have unsanitised issue codes

srijan-deepsource opened the pull request:
Here's how test failures are gonna look:

<img width="836" alt="image" src="https://github.com/DeepSourceCorp/community-analyzers/assets/68371686/e48f1db3-7b74-4f50-b47c-478b4b34f231">

srijan-deepsource committed:
chore: fail tests if the test sarif reports have unsanitised issue codes

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in e389428 according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/14

eshaan-deepsource commented on tests/test_report_parsing.py:
```diff
@@ -46,7 +71,22 @@ def test_sarif_parser() -> None:
+            missing_ids = []
             assert result
             assert result["issues"] is not None
+            for issue in result["issues"]:
+                issue_code = issue["issue_code"]
+                if issue_code not in active_issues:
+                    missing_ids.append(issue_code)
```
```python3
missing_ids = set(issue["issue_code"] for issue in result["issues"]) - active_issues
```

I believe this can be written in a much cleaner way like this. Also because `active_issues` is already a set.

tushar-deepsource replied:
This will be much faster as well.

tushar-deepsource replied:
though you can use set comprehension
```suggestion
            missing_ids = {issue["issue_code"] for issue in result["issues"]} - active_issues
```

srijan-deepsource replied:
Right. Missed it. Was generating a issue code to id mapping first. Didn't look at it again when I changed the dictionary to set. 

Good catch!

srijan-deepsource committed:
Update test_report_parsing.py

Co-authored-by: Tushar Sadhwani <86737547+tushar-deepsource@users.noreply.github.com>
Signed-off-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>

deepsource-autofix[bot] committed:
style: format code with Black and isort

This commit fixes the style issues introduced in 9a3e29a according to the output
from Black and isort.

Details: https://github.com/DeepSourceCorp/community-analyzers/pull/14

srijan-deepsource committed:
fix issue flagged by mypy

srijan-deepsource commented on tests/test_report_parsing.py:
```diff
@@ -46,7 +71,21 @@ def test_sarif_parser() -> None:
         report_path = Path(reports_base_dir, report).as_posix()
+
         with parse_single_artifact(report_path, report_tool) as result:
+            missing_ids = set()
```
```suggestion
```

srijan-deepsource committed:
unnecessary assignment

Signed-off-by: Srijan Saurav <68371686+srijan-deepsource@users.noreply.github.com>
